### ISSUE TITLE
Map cache filename with many ignore_hosts exceeds NAME_MAX

### ISSUE THREAD

**Issue by miragu** [NagVis/nagvis#49]
Hello,
When using automaps with a certain amount of FQDN hostnames in ignore_hosts I noticed that the cache creation would fail (can't remember the specific error but something regarding fopen, filename too long). In effect, it easily exceeds NAME_MAX (255 under POSIX; not to be confused with PATH_MAX which is usually 4096). My fix was to simply remove ignore_hosts from the cache filename. Not sure if it causes other problems. I noticed you already remove source_file and have a FIXME to configure these exceptions. Anyhow, here's a trivial patch that fixed my problem.
[nagvis-cache-filename-fix.txt](https://github.com/NagVis/nagvis/files/237022/nagvis-cache-filename-fix.txt)

**Comment by LarsMichelsen** [comment]
Yes. This naming scheme is a bad idea. It should be rewritten to use some hashing algorithm for generating a unique id based on the parameters. The parameters should be kept somewhere to make debugging easier. e.g. in a separate file next to the cache file.
